travis/install_grpc.sh

```diff
@@ -12,19 +12,19 @@ if [ "$grpc_dist" != "" ]; then
 fi
 
 # for python, we'll need the latest virtualenv and tox.
-# running gRPC requires the six package, any version will do.
+# running gRPC requires the six package, version >=1.10.
 if [ "$grpc_dist" != "" ]; then
-  pip install --upgrade --root $grpc_dist --ignore-installed virtualenv tox
-  pip install --root $grpc_dist --ignore-installed six
+  # Create a virtualenv, which also creates a virualenv-boxed pip.
+  virtualenv $grpc_dist/usr/local
+  $grpc_dist/usr/local/bin/pip install --upgrade --ignore-installed virtualenv tox six
```

Comment:
No, I did it because it's the only way to get pip to see the updated `six` version. If you already have an older version of `six` installed via apt (likely a dependency of something else you installed, so you can't remove just python-six) then running regular pip will fail because it prefers the older `six` in system-wide dist-packages even though there's a newer one in /usr/local site-packages.

To force it to use the new one and ignore the system-wide one, you have to use virtualenv. I'm running into this same problem while trying to build the Docker bootstrap image. There we try to install grpc as a system package, but that's not possible because there's an older system python-six installed that's conflicting. Still looking for workarounds...